DEVELOPING.md

```diff
@@ -25,6 +27,119 @@ are shared between multiple packages then they should be added to the Yarn const
 This is also where we define common peerDependencies and enforce a common version. These are packages such as React Native
 that our packages require but that we leave to the customer to have defined as explicit dependencies.
 
+## Adding new Native Modules
```

Comment:
this has some similar information to `packages/react-native-tracer-provider/DEVELOPING.md` but is more comprehensive, I added a note to https://www.notion.so/embraceio/Strategy-Template-for-new-packages-that-have-native-code-10a7e3c99852800faf13cc70fe783f25?pvs=4 that after merging we can delete `packages/react-native-tracer-provider/DEVELOPING.md` and move any extra information from there into this file